TrkDiag/src/BkgDiag_module.cc

```diff
@@ -201,36 +210,31 @@ namespace mu2e
       throw cet::exception("RECO")<<"mu2e::BkgDiag: data inconsistent"<< endl;
     // loop over background clusters
 
-
     _nhits=0;
+    _hitPos.clear();
+    _hitPos.reserve(_chcol->size());
     for(size_t ich=0;ich<_chcol->size();++ich){
-      _hitRad[_nhits]    = sqrtf(_chcol->at(ich).pos().perp2());
-      _hitPhi[_nhits]    = _chcol->at(ich).pos().phi();
-      _hitHPhi[_nhits]   = _chcol->at(ich).helixPhi();
+      _hitPos.push_back(_chcol->at(ich).pos());
```

Comment:
To be honest, Radius, Phi and Z is more useful than X,Y and Z, but we could go this way if you want.